lib/BoundaryValueDiffEqCore/src/utils.jl

```diff
@@ -370,3 +370,7 @@ end
 @inline (f::__Fix3{F})(a, b) where {F} = f.f(a, b, f.x)
 
 # convert every vector of vector to AbstractVectorOfArray, especially if them come from get_tmp of PreallocationTools.jl
+
+get_dense_ad(::Nothing) = nothing
+get_dense_ad(ad) = ad
+get_dense_ad(ad::AutoSparse) = ADTypes.dense_ad(ad)
```

Comment:
what is this for?

Reply:
you seem to use it everywhere but if you go back from `AutoSparse(backend, ...)` to `backend` you lose the benefits of sparse AD

Reply:
Yeah, should use the dense ad from diffmode, just fixed